datashader/layout.py

```diff
@@ -171,7 +171,7 @@ def _merge_points_with_nodes(nodes, points, params):
     return n
 
 
-@nb.jit(nogil=True)
+@nb.jit(forceobj=True)
```

Comment:
Uses numpy array creation and cannot therefore be compiled in nopython mode, could maybe be rewritten?

Reply:
To expand a little bit here, the numpy array creation isn't really the problem. The problem is that looplifting also isn't possible due to some type inference issues and presumably the ``.toarray()`` call inside the loop.

Reply:
According to Siu, the problem is that `matrix` is a scipy.sparse.csr.csr_matrix, and Numba can't yet speed up sparse matrices.

Reply:
Fixed in https://github.com/pyviz/datashader/pull/763 .